pkg/agent/openflow/network_policy.go

```diff
@@ -150,7 +150,30 @@ type conjunctiveMatch struct {
 }
 
 func (m *conjunctiveMatch) generateGlobalMapKey() string {
-	return fmt.Sprintf("table:%d,type:%d,value:%s", m.tableID, m.matchKey, m.matchValue)
+	var valueStr string
+	matchType := m.matchKey
+	switch m.matchValue.(type) {
+	case net.IP:
+		// Use the unique format "x.x.x.x/xx" for IP address and IP net, to avoid generating two different global map
+		// keys for IP and IP/32. Use MatchDstIPNet/MatchSrcIPNet as match type to generate global cache key for both IP
+		// and IPNet. This is because OVS treats IP and IP/32 as the same condition, if Antrea has two different
+		// conjunctive match flow contexts, only one flow entry is installed on OVS, and the conjunctive actions in the
+		// first context will be overwritten by those in the second one.
+		ip, _ := m.matchValue.(net.IP)
```

Comment:
Use `switch value := m.matchValue.(type)`, then you don't have to do another type assertion here and L171, `value` will be the target type for your usage: https://tour.golang.org/methods/16

Reply:
Appriatiate for this suggesion, I will update.